File: app/src/main/java/com/denchic45/kts/utils/Dimensions.java
package com.denchic45.kts.utils;

import android.content.Context;
import android.util.DisplayMetrics;

import org.jetbrains.annotations.NotNull;

public final class Dimensions {

    public static int dpToPx(int dp, @NotNull Context context) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        return Math.round(dp * (displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT));
    }
}

File: app/common/src/androidMain/kotlin/com/denchic45/studiversity/util/Dimensions.kt
package com.denchic45.studiversity.util

import android.content.Context
import android.util.DisplayMetrics
import kotlin.math.roundToInt

object Dimensions {
    fun dpToPx(dp: Int, context: Context): Int {
        val displayMetrics = context.resources.displayMetrics
        return (dp * (displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT)).roundToInt()
    }
}

 fun Context.dpToPx(dp: Int) = Dimensions.dpToPx(dp, this)